test/common.py

```diff
@@ -403,6 +425,44 @@ def run_train_translate(train_params: str,
                             # for negative constraints, ensure the constraints is *not* in the constrained output
                             assert restriction not in constrained_out
 
+
+        # Test scoring. We make sure that we can score the (input, translation output) and get the same
+        # model score.
+        if not use_prepared_data:
+            ## Score
+            # We use the translation parameters, but have to remove irrelevant arguments from it.
+            # Currently, the only relevant flag passed is the --softmax-temperature flag.
+            score_params = ''
+            if 'softmax-temperature' in translate_params:
+                tokens = translate_params.split(C.TOKEN_SEPARATOR)
+                for i, token in enumerate(tokens):
+                    if token == '--softmax-temperature':
+                        score_params = '--softmax-temperature {}'.format(tokens[i + 1])
+                        break
+
+            scores_output_file = out_path + '.score'
+            params = "{} {} {}".format(sockeye.score.__file__,
+                                       _SCORE_PARAMS_COMMON.format(model=model_path,
+                                                                   source=test_source_path,
+                                                                   target=out_path,
+                                                                   output=scores_output_file),
+                                       score_params)
+
+            if test_source_factor_paths is not None:
+                params += _SCORE_WITH_FACTORS_COMMON.format(source_factors=" ".join(test_source_factor_paths))
+
+            with patch.object(sys, "argv", params.split()):
+                sockeye.score.main()
+
+            ## Compare scored output to original translation output. First remove -inf lines from the translate_score_path
+            ## file. These correspond to blank lines in translate, which are skipped in sockeye.score.
+            for translate_score, score_score in zip(filter(lambda x: x != '-inf\n', open(translate_score_path).readlines()),
+                                                    open(scores_output_file).readlines()):
+                translate_score = float(translate_score)
+                score_score = float(score_score)
+                print('SCORES', translate_score, score_score)
+                assert abs(translate_score - score_score) < 0.002
```

Comment:
just 0.002 ? Do we except such large diffs?

Reply:
I find values often differ on the third point to the right of the decimal (e.g., 1.567 versus 1.568). Testing for `<= 0.001` doesn't cover this case.